packages/core-utils/src/batches.ts

```diff
@@ -4,6 +4,19 @@ import {
   TransactionResponse,
 } from '@ethersproject/abstract-provider'
 
+export interface RollupInfo {
+  mode: 'sequencer' | 'verifier'
+  syncing: boolean
+  ethContext: {
+    blockNumber: number
+    timestamp: number
+  }
+  rollupContext: {
+    index: number
+    queueIndex: number
+  }
+}
```

Comment:
I'm not completely sure if I agree with migrating this type into `core-utils`. It seems to me like it's only used in once place, the `batch-submitter`, in a very specific way. Things in `core-utils` should probably satisfy one of two criteria:

1. Used by multiple services in a meaningful way
2. Is highly coupled to the contracts implementation / encoding structures

I feel like this is neither? Would love to understand the motivation better.

Reply:
@snario this is information you get by calling `rollup_getInfo` so it _is_ useful for anyone consuming Optimism, seems useful to me.